app/services/idv/steps/upload_step.rb

```diff
@@ -63,7 +63,11 @@ def bypass_send_link_steps
       end
 
       def mobile_device?
```

Comment:
As mentioned in Slack, I think we should just remove this mobile device handling from this step, rather than continuing to prop it up. If I had to guess, it's a remnant from when we supported no-JavaScript for identity verification, since our client-side mobile checks wouldn't have assigned the `:skip_upload_step` form value. Currently, we should not have any mobile users at this step.

Reply:
Unless I've misunderstood something, the iPad was reaching this upload step (and updating the mobile check was the only way I found to get the flow to the capture page).
  
I think what's happening is that the [frontend sets](https://github.com/18F/identity-idp/blob/main/app/javascript/packs/document-capture-welcome.js#L17) the `skip_upload` hidden input, which the agreement step picks up [here](https://github.com/18F/identity-idp/blob/main/app/services/idv/steps/agreement_step.rb#L10), which then sets the session's `:skip_upload_step` to true [here](https://github.com/18F/identity-idp/blob/main/app/services/idv/steps/agreement_step.rb#L15). 
  
I'm still wrapping my head around FSM, but does that somehow mean that the upload step should be skipped entirely, or that once it is entered some logic checks for `:skip_upload_step` and then moves on? (It seems like the upload step is always a part of the [flow](https://github.com/18F/identity-idp/blob/main/app/services/idv/flows/doc_auth_flow.rb#L7) right after the agreement)

Reply:
The way I expect it to work is that we segment users as "mobile" and "non-mobile" users, and only the non-mobile users should see the upload step.

To your point, previously we weren't considering iPad as mobile, and so it would see this step. It sounds like we want to treat it as mobile, so it should skip this screen.

This check should happen once, in `document-capture-welcome.js`, and we shouldn't need to consider it in this step's code, since we trust that this step will be skipped entirely.

The missing piece of the logic is here:

https://github.com/18F/identity-idp/blob/23507813c6c82d14e0a2277d4e47b61a49ec6a18/app/controllers/idv/doc_auth_controller.rb#L48-L52

Ideally, we'd just call `mark_step_complete(:upload)` in the agreement step's `skip_to_capture`. If I recall correctly, it was done this way primarily so that we'd still be logging that the upload step was visited for mobile users, for funnel data integrity.

Reply:
Ah, I think you're right, in that we rely on this step's `call` to have been called even for mobile users who will be skipping it entirely (via [`DocAuthController#update_if_skipping_upload`](https://github.com/18F/identity-idp/blob/23507813c6c82d14e0a2277d4e47b61a49ec6a18/app/controllers/idv/doc_auth_controller.rb#L48-L52)). In that case, checking `skip_upload_step` as an indicator of a mobile device seems like a good idea. I'd almost wonder if we should _exclusively_ check that value, since we trust that it would be set by the client-side code, and we'd not want to deal with maintaining two different mobile device checks. 

I still think we should get rid of [the mobile-to-desktop "send user to email sent" flow](https://github.com/18F/identity-idp/blob/23507813c6c82d14e0a2277d4e47b61a49ec6a18/app/services/idv/steps/upload_step.rb#L18-L20), but if we're not dealing with removing `mobile_device?`, I think we could defer that to a separate ticket.

Reply:
> I'd almost wonder if we should exclusively check that value, since we trust that it would be set by the client-side code, and we'd not want to deal with maintaining two different mobile device checks
  
I agree. There a sort-of-related issue as well: our server-side gem is likely not logging iPads (and possibly android tablets) if its relying on user agents. That means we have no real view into who our iPad users are, and therefore how many of them might have been affected by not having Acuant capture available. Which begs the question -- should we be logging something here, too?
  
> I still think we should get rid of [the mobile-to-desktop "send user to email sent" flow](https://github.com/18F/identity-idp/blob/23507813c6c82d14e0a2277d4e47b61a49ec6a18/app/services/idv/steps/upload_step.rb#L18-L20), but if we're not dealing with removing mobile_device?, I think we could defer that to a separate ticket.
  
Makes sense. Do you happen to know where the `params[:type]` is being set to `'desktop'`?

Reply:
> Do you happen to know where the `params[:type]` is being set to `'desktop'`?

That's a good question! It was tricky to track down, but I eventually found it. It's the query parameter being assigned here in the form action:

https://github.com/18F/identity-idp/blob/75c5dbb40863ccb4ed24be6eabcd2c181f766939/app/views/idv/doc_auth/upload.html.erb#L66